spec/acceptance/reporting_spec.rb

```diff
@@ -56,116 +56,74 @@
+    include_examples 'ictc', report_label: 'no_changes', noop_test: true
+  end
 
-    context 'with report status: corrective changes' do
-      include_context 'incident creation test setup'
-      include_context 'corrective change'
-      include_examples 'incident creation test', 'changed'
+  context "when the incident creation condition includes the 'always' condition" do
+    let(:params) do
+      super().merge('incident_creation_conditions' => ['always'])
     end
 
-    context 'with report status: failed changes' do
-      let(:sitepp_content) do
-        to_manifest(declare('exec', 'foo', 'command' => '/bin/foo_command'))
-      end
-
-      include_context 'incident creation test setup'
-      include_examples 'incident creation test', 'failed'
-    end
+    include_examples 'ictc', report_label: 'failures'
+    include_examples 'ictc', report_label: 'corrective_changes'
+    include_examples 'ictc', report_label: 'intentional_changes'
+    include_examples 'ictc', report_label: 'pending_intentional_changes'
+    include_examples 'ictc', report_label: 'pending_corrective_changes'
+    include_examples 'ictc', report_label: 'no_changes'
   end
 
-  context 'with user-specified incident_creation_report_statuses' do
-    context 'with pending changes reporting enabled' do
-      let(:params) do
-        super().merge('incident_creation_conditions' => ['corrective_changes', 'pending_changes'])
-      end
-
-      context 'with a pending corrective change' do
-        include_context 'incident creation test setup'
-        include_context 'corrective change', true
-        include_examples 'incident creation test', 'pending'
-      end
+  context "when incident_creation_conditions == ['never']" do
+    let(:params) do
+      super().merge('incident_creation_conditions' => ['never'])
     end
 
-    context 'with unchanged reporting enabled' do
-      let(:params) do
-        super().merge('incident_creation_conditions' => ['no_changes'])
-      end
-
-      context 'with a an unchanged run' do
-        let(:sitepp_content) do
-          # Puppet should report that nothing happens
-          ''
-        end
-
-        include_context 'incident creation test setup'
-        include_examples 'incident creation test', 'unchanged'
-      end
-    end
+    include_examples 'ictc', report_label: 'failures', noop_test: true
+    include_examples 'ictc', report_label: 'corrective_changes', noop_test: true
+    include_examples 'ictc', report_label: 'intentional_changes', noop_test: true
+    include_examples 'ictc', report_label: 'pending_intentional_changes', noop_test: true
+    include_examples 'ictc', report_label: 'pending_corrective_changes', noop_test: true
+    include_examples 'ictc', report_label: 'no_changes', noop_test: true
+  end
 
-    context 'with no attributes selected ([])' do
+  # This is testing a bugfix from a previous module version
+  context 'distinguishing intentional changes from corrective changes' do
```

Comment:
This was a bug in the previous PR.

I think the other bugs (distinguishing pending intentional vs. pending corrective changes) are covered in the user-specified tests.